server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/VacationRoutesTest.java

```diff
@@ -0,0 +1,257 @@
+        .build();
+
+    private static String isoString(ZonedDateTime date) {
+        return date.format(DateTimeFormatter.ISO_DATE_TIME);
+    }
+
+    private WebAdminServer webAdminServer;
+    private MemoryVacationRepository vacationRepository;
+    private MemoryNotificationRegistry notificationRegistry;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        vacationRepository = new MemoryVacationRepository();
+        notificationRegistry = new MemoryNotificationRegistry(new DefaultZonedDateTimeProvider());
+
+        SimpleDomainList domainList = new SimpleDomainList();
+        domainList.addDomain(DOMAIN);
+        MemoryUsersRepository usersRepository = MemoryUsersRepository.withVirtualHosting(domainList);
+
+        VacationRoutes vacationRoutes = new VacationRoutes(
+            vacationRepository, notificationRegistry, usersRepository, new JsonTransformer());
+        this.webAdminServer = WebAdminUtils.createWebAdminServer(vacationRoutes).start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer).build();
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+
+        usersRepository.addUser(Username.of(BOB), "secret");
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void getVacation() {
+        vacationRepository.modifyVacation(AccountId.fromString(BOB), VacationPatch.builderFrom(VACATION).build()).block();
+
+        when()
+            .get(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body("enabled", equalTo(VACATION.isEnabled()))
+            .body("fromDate", equalTo(isoString(VACATION.getFromDate().get())))
+            .body("toDate", equalTo(isoString(VACATION.getToDate().get())))
+            .body("subject", equalTo(VACATION.getSubject().get()))
+            .body("textBody", equalTo(VACATION.getTextBody().get()))
+            .body("htmlBody", equalTo(VACATION.getHtmlBody().get()));
+    }
+
+    @Test
+    void getVacationFails() {
+        when()
+            .get(VacationRoutes.VACATION + SEPARATOR + CAROL)
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+            .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+            .body("message", is("The user 'carol@example.org' does not exist"));
+    }
+
+    @Test
+    void postVacationCreates() throws Exception {
+        AccountId bob = AccountId.fromString(BOB);
+        RecipientId alice = RecipientId.fromMailAddress(new MailAddress(ALICE));
+
+        vacationRepository.modifyVacation(bob, VacationPatch.builderFrom(VACATION).build())
+                .then(notificationRegistry.register(bob, alice, Optional.empty())).block();
```

Comment:
With this isn't this test technically an update, like the below test? Shouldn't we remove this instead to really test a creation from nothing?

Reply:
It is a misnomer here, I actually wanted to test that all fields can be updated. But you are right, creation from scratch is also a good test case, so I added an extra one. There now is Create, UpdatesAll, UpdatesPartial.